sql: remove catalog physical accessors

thoszhang opened the pull request:
This is a small first step toward having `descs.Collection` be responsible for all descriptor accesses. See individual commits for details.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55625)
<!-- Reviewable:end -->

Commit:
sql/catalog: fix catalogkv accessor function

This commit amends `catalogkv.MustGetSchemaByID` to correctly return
an error when the schema is not found instead of returning nil. There
should be no user-facing change because the current callers of the
function aren't susceptible to the problem.

Release note: None

Commit:
sql/catalog: remove UncachedPhysicalAccessor from test_utils

This commit removes `UncachedPhysicalAccessor` usages from test_utils
and replaces them with calls to catalogkv functions. It also removes it
from an unrelated sql test.

Release note: None

Commit:
sql: use descs.Txn in TemporaryObjectCleaner

This replaces the use of `UncachedPhysicalAccessor` and other direct
read and writes to/from the store.

Release note: None

Commit:
sql/catalog: remove physical accessors

This commit removes the `CachedPhysicalAccessor` and converts the
`UncachedPhysicalAccessor` methods into unexported functions in `descs`
(with the exception of `GetObjectNames`, which will be dealt with later).

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> this is a nice step forward. Not sure what's going on with the build.

> Reviewed 1 of 1 files at r1, 2 of 2 files at r2, 4 of 4 files at r3, 9 of 9 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55625#-:-MKjaYhWCVSmVMZ_ZvJ3:b1510ec)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @lucy-zhang)

___
*[pkg/sql/catalog/descs/physical_accessor.go, line 159 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/55625#-MKjaGTq8ZtqnpUfgdbi:-MKjaGTq8ZtqnpUfgdbj:bksz7ly) ([raw file](https://github.com/cockroachdb/cockroach/blob/d61462fafe0e28d57c9baa269d6a82b59df64d3b/pkg/sql/catalog/descs/physical_accessor.go#L159)):*
> ```
> // TODO (lucy): This is exported as a standalone function for now, but it should
> // either go on descs.Collection or not exist.
> ```

I can see an argument to move it to `catalogkv` for now and thus drive removing access to it in the push to remove access to catalogkv


<!-- Sent from Reviewable.io -->

thoszhang reviewed (commented):
I had to regenerate the bazel build files. Should be fine now.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55625#-:-MKk2Gy6--qaXSs9u2l1:b-webtso)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @ajwerner)

___
*[pkg/sql/catalog/descs/physical_accessor.go, line 159 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/55625#-MKjaGTq8ZtqnpUfgdbi:-MKk2bMY51pRkJsRvLCi:b4kyjx5) ([raw file](https://github.com/cockroachdb/cockroach/blob/d61462fafe0e28d57c9baa269d6a82b59df64d3b/pkg/sql/catalog/descs/physical_accessor.go#L159)):*
<details><summary><i>Previously, ajwerner wrote…</i></summary><blockquote>

> ```
> // TODO (lucy): This is exported as a standalone function for now, but it should
> // either go on descs.Collection or not exist.
> ```

I can see an argument to move it to `catalogkv` for now and thus drive removing access to it in the push to remove access to catalogkv
</blockquote></details>

I tried to keep it there, but it seemed like more trouble than it was worth since it calls `descs.getSchema()` (which itself is nontrivial, since we depend on its knowledge of the public and virtual schemas). I added a note to remind myself.


<!-- Sent from Reviewable.io -->

lucy-zhang commented:
TFTR

bors r+
